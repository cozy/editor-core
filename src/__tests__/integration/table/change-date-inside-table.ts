import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';

import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
  insertBlockMenuItem,
} from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';

const calendar = '[aria-label="calendar"]';
const nextDate = 'td[aria-selected=true] + td';

BrowserTestCase(
  'change-date-inside-table.ts: Change date inside table',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowDate: true,
      allowTables: {
        advanced: true,
      },
    });

    await page.click(editable);

    // Insert table
    await quickInsert(page, 'Table');
    // Insert Date
    await insertBlockMenuItem(page, 'Date');
    expect(await page.isExisting(calendar)).toBe(true);

    await page.click(nextDate);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
