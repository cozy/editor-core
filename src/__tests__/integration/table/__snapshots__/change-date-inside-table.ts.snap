// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change-date-inside-table.ts: Change date inside table 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "timestamp": "1502928000000",
                      },
                      "type": "date",
                    },
                    Object {
                      "text": " ",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;
